### PR TITLE
switch to scimloperators

### PR DESCRIPTION
**Pull request by vpuri3**
https://github.com/SciML/SciMLOperators.jl/issues/55

### CODE REVIEW

**Comment by ChrisRackauckas** [comment]
This is missing the deprecation paths. At least https://github.com/SciML/SciMLBase.jl/pull/183 needs to be in here.

**Comment by vpuri3** [comment]
vector diagonal preconditioners should be switched as follows:
```julia
            s = rand(n)
            #Pl, Pr = DiagonalOperator(s), LinearSolve.InvPreconditioner(Diagonal(s))
            Pl = DiagonalOperator(s)
            Pr = inv(Pl)

            Pl = cache_operator(Pl, x)
            Pr = cache_operator(Pr, x)
```

**Comment by vpuri3** [comment]
ok lemme do that. unrelated, should we require `LinearProblem.A` to be a `SciMLOperator`?

**Comment by ChrisRackauckas** [comment]
No. We should allow it but not require it.

File: src/problems/basic_problems.jl [hunk 1]
```diff
@@ -67,8 +67,8 @@ struct LinearProblem{uType, isinplace, F, bType, P, K} <:
 end
 
 function LinearProblem(A, b, args...; kwargs...)
-    if A isa AbstractArray
-        LinearProblem{true}(A, b, args...; kwargs...)
+    if A isa AbstractMatrix
+        LinearProblem{true}(MatrixOperator(A), b, args...; kwargs...)
```
Comment (ChrisRackauckas) [inline src/problems/basic_problems.jl:71]: keep it simple

Comment (vpuri3) [reply src/problems/basic_problems.jl:71]: ok

**Comment by vpuri3** [comment]
anything else to add chris?

**Comment by vpuri3** [comment]
what does this error mean @ChrisRackauckas 
```julia
[3f19e933] + p7zip_jll
    Updating registry at `~/.julia/registries/General`
    Updating git-repo `[https://github.com/JuliaRegistries/General.git`](https://github.com/JuliaRegistries/General.git%60)
  No Changes to `~/work/SciMLBase.jl/SciMLBase.jl/downstream/Project.toml`
  No Changes to `~/work/SciMLBase.jl/SciMLBase.jl/downstream/Manifest.toml`
ERROR: LoadError: KeyError: key UUID("7ed4a6bd-45f5-4d41-b270-4a48e9bafcae") not found
Stacktrace:
  [1] getindex
    @ ./dict.jl:481 [inlined]
  [2] getindex
    @ /opt/hostedtoolcache/julia/1.7.3/x64/share/julia/stdlib/v1.7/Pkg/src/Types.jl:273 [inlined]```

**Comment by ChrisRackauckas** [comment]
Did you find/replace and change the name in the Project.toml? Is SciMLOperators in the Project.toml now pointing to the wrong UUID?

**Comment by vpuri3** [comment]
do deprecations as follows:
```julia
const DiffEqArrayOperator = MatrixOperator

function DiffEqArrayOperator(args...)
@warn "SciMLBase.DiffEqArrayOperator is deprecated.
Use SciMLOperators.MatrixOperator instead"

MatrixOperator(args...)
end
```

File: .github/workflows/Downstream.yml [hunk 2]
```diff
@@ -28,13 +28,10 @@ jobs:
           - {user: SciML, repo: OrdinaryDiffEq.jl, group: Regression}
           - {user: SciML, repo: StochasticDiffEq.jl, group: All}
           - {user: SciML, repo: Sundials.jl, group: All}
-          - {user: SciML, repo: LinearSolve.jl, group: All}
           - {user: SciML, repo: NonlinearSolve.jl, group: All}
           - {user: SciML, repo: Optimization.jl, group: All}
           - {user: SciML, repo: Integrals.jl, group: All}
-          - {user: SciML, repo: DiffEqSensitivity.jl, group: All}
           - {user: SciML, repo: ModelingToolkit.jl, group: All}
-          - {user: SciML, repo: DiffEqOperators.jl, group: All}
```
Comment (ChrisRackauckas) [inline .github/workflows/Downstream.yml]: These shouldn't be removed.

**Comment by vpuri3** [comment]
i think this is the error showing up in downstream tests:

https://github.com/JuliaLang/Pkg.jl/issues/2776

**Comment by vpuri3** [comment]
the problem is the SciMLOperators depends on DiffEqBase.

**Comment by vpuri3** [comment]
nonlinearsolve error should be fixed by https://github.com/SciML/NonlinearSolve.jl/pull/90

**Comment by vpuri3** [comment]
The only test failing for OrdinaryDiffEq is 
```julia
if VERSION >= v"1.8"
    @test 2t1 < t3 # FAILING
    @test t2 < t4
    integ = init(lorenzprob, Rosenbrock23())
    @test integ.f.f isa SciMLBase.FunctionWrappersWrappers.FunctionWrappersWrapper
end
```
in `OrdinaryDiffEq/test/interface/norecompile`. And this test is failing on master as well so this doesn't seem like an issue with this branch.

**Comment by vpuri3** [comment]
https://github.com/SciML/OrdinaryDiffEq.jl/pull/1766 should fix error in OrdinaryDiffEq / Integrators / 1 test error

**Comment by vpuri3** [comment]
[ci-skip]

**Comment by vpuri3** [comment]
some of these workflows have been stuck for ~50 mins. closing this PR for a moment so workflows can start on other PR

**Comment by vpuri3** [comment]
closing this so workflows don't run. need https://github.com/SciML/OrdinaryDiffEq.jl/pull/1766 first

**Comment by vpuri3** [comment]
retriggering CI after ordinarydiffeq merge

**Comment by vpuri3** [comment]
good, previous errors are resolved. :D 

new error: looks like we're converting a sparse matrix into a dense one here

```julia
Linear Solver Tests: Error During Test at /home/runner/.julia/packages/SafeTestsets/A83XK/src/SafeTestsets.jl:25
  Got exception outside of a @test
  LoadError: MethodError: no method matching KLU.KLUFactorization(::Matrix{Float64})
  Closest candidates are:
    KLU.KLUFactorization(::Any, ::Any, ::Any, ::Any) at ~/.julia/packages/KLU/JK7t1/src/KLU.jl:187
    KLU.KLUFactorization(::SparseArrays.SparseMatrixCSC{Tv, Ti}) where {Tv<:Union{Float64, ComplexF64}, Ti<:Union{Int32, Int64}} at ~/.julia/packages/KLU/JK7t1/src/KLU.jl:221
  Stacktrace:
    [1] init_cacheval(alg::LinearSolve.KLUFactorization, A::Matrix{Float64}, b::Vector{Float64}, u::Vector{Float64}, Pl::LinearSolve.InvPreconditioner{LinearAlgebra.Diagonal{Float64, Vector{Float64}}}, Pr::LinearAlgebra.Diagonal{Float64, Vector{Float64}}, maxiters::Int64, abstol::Float64, reltol::Float64, verbose::Bool, assumptions::LinearSolve.OperatorAssumptions{Nothing})
      @ LinearSolve ~/.julia/packages/LinearSolve/tyXtB/src/factorization.jl:311
 ```

**Comment by ChrisRackauckas** [comment]
This will not be accepted without a clean diff. There seems to be a lot broken in here too, I'm not convinced there aren't merge issues.

**Pull request closed without merging**